homeassistant/components/ring/__init__.py

```diff
@@ -40,7 +53,10 @@ def setup(hass, config):
         ring = Ring(username=username, password=password, cache_file=cache)
         if not ring.is_connected:
             return False
-        hass.data['ring'] = ring
+        hass.data[DATA_RING_CHIMES] = ring.chimes
+        hass.data[DATA_RING_DOORBELLS] = ring.doorbells
+        hass.data[DATA_RING_STICKUP_CAMS] = ring.stickup_cams
```

Comment:
If we cache the devices in local variables at the same time as storing them in hass.data we can perhaps avoid breaking the line below when adding them into `ring_devices`.
```py
hass.data[DATA_RING_CHIMES] = chimes = ring.chimes
...
```

Reply:
think I've done this the way you suggested